### PR TITLE
Stt adapter IBM - andrew d anderson

### PR DESCRIPTION
**Pull request by pietrop**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      

from original PR https://github.com/bbc/react-transcript-editor/pull/60 by @AndrewDAnderson

**Describe what the PR does**    

Adds support for IBM Watson STT (cc @seanmtracey)

**State whether the PR is ready for review or whether it needs extra work**    

Updated PR https://github.com/bbc/react-transcript-editor/pull/60 with current master, and doing some final tweaks before merging to master, ahead off reorg (PR https://github.com/bbc/react-transcript-editor/pull/122 - issue https://github.com/bbc/react-transcript-editor/issues/109) 

**Additional context**    
NA

### CODE REVIEW

**Comment by pietrop** [comment]
Need to double check the tests are passing, but other then that ready for merge to master.

cleaned up the code making it more in line with the logic in the other adapter, and better setup for accommodating changes and adjustment over time.

Seems like IBM speaker diarization might actually be quiet good, in this example it recognised it as being just one speaker. 

So rather then grouping by speaker, which would have left with one giant paragraph and indured draftjs performance, with virtual dom etc.. I left some comments in the code on where that could be changed, eg it could break paragraphs every minute.

This also loses some of the clever stuff that was done in the previous PR about adding punctuation to IBM STT which doesn't seem to have it (?).

[Seems like a common workaround is to add punctuation after every IBM line](https://developer.ibm.com/answers/questions/322036/watson-speech-to-text-punctuation-1/) but will leave that for a later PR.

**Comment by pietrop** [comment]
fixed the test, ready for merge 

Screen Shot 2019-03-20 at 19 49 20

File: src/lib/Util/adapters/ibm/index.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  };
+
+  const ibmNormalisedWordsToDraftJs = (ibmNormalisedWordsWithSpeakers) => {
+    const drafJsParagraphsResults = [];
+    ibmNormalisedWordsWithSpeakers.forEach((ibmParagraph) => {
+      const draftJsContentBlockParagraph = {
+        text: ibmParagraph.map((word) => {return word.text;}).join(' '),
+        type: 'paragraph',
+        data: {
+          // TODO: Assuming each paragraph in IBM line is the same
```
Comment (jamesdools) [inline src/lib/Util/adapters/ibm/index.js]: does more work need to be done here?

File: src/lib/Util/adapters/ibm/index.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
+/**
```
Comment (jamesdools) [inline src/lib/Util/adapters/ibm/index.js]: Let's remove these comments - we have some sample json files to refer to if needed.

File: src/lib/Util/adapters/ibm/index.js [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+        word.speaker = findSpeakerSegmentForWord(word, ibmSpeakers);
+
+        return word;
+      });
+    });
+  };
+
+  const ibmNormalisedWordsToDraftJs = (ibmNormalisedWordsWithSpeakers) => {
+    const drafJsParagraphsResults = [];
```
Comment (jamesdools) [inline src/lib/Util/adapters/ibm/index.js]: let's just correct this variable name.

File: src/lib/Util/adapters/ibm/index.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+      };
+      drafJsParagraphsResults.push(draftJsContentBlockParagraph);
+    });
+
+    return drafJsParagraphsResults;
+  };
+
+  const normalisedWords = normalizeIBMWordsList(ibmJson.results[0].results);
+  // nested array of words, to keep some sort of paragraphs, in case there's only one speaker
+  // can be refactored/optimised later
```
Comment (jamesdools) [inline src/lib/Util/adapters/ibm/index.js]: Will it? Let's make this comment a `TODO:`'s and actually follow up on it!

Slightly side note but there's lots of comments like this in the code and it gets lost. Even though there are a fair few TODOs too 😅

File: src/lib/Util/adapters/ibm/index.js (hunk 1 above)
Comment (pietrop) [reply src/lib/Util/adapters/ibm/index.js]: Don't think so, It just seems like the IBM data structure gives you word level speakers,but also gives you "lines" so assuming each word in a line has the same speaker.

Can also check with @seanmtracey if has time to review this later.

Comment (pietrop) [reply src/lib/Util/adapters/ibm/index.js]: removing the todo and expanding on the comment to give more context.

File: src/lib/Util/adapters/ibm/index.js (hunk 3 above)
Comment (pietrop) [reply src/lib/Util/adapters/ibm/index.js]: well spotted

File: src/lib/Util/adapters/ibm/index.js (hunk 4 above)
Comment (pietrop) [reply src/lib/Util/adapters/ibm/index.js]: ok ,fair added the TODO

File: src/lib/Util/adapters/ibm/index.js (hunk 2 above)
Comment (pietrop) [reply src/lib/Util/adapters/ibm/index.js]: ok, removed across all the other adapters as well

**Pull request merged**